avro-converter/src/test/java/io/confluent/connect/avro/AvroDataTest.java

```diff
@@ -278,10 +281,133 @@ public void testFromConnectComplex() {
+
+    org.apache.avro.Schema mapSchema = org.apache.avro.SchemaBuilder.builder().map().values().intType();
+    ObjectNode mapNode = JsonNodeFactory.instance.objectNode();
+    mapNode.put("field", 1);
+    mapSchema.addProp("connect.default", mapNode);
+
+    org.apache.avro.Schema nonStringMapSchema = org.apache.avro.SchemaBuilder.builder().array()
+                                                                             .items(complexMapElementSchema);
+    ArrayNode nonStringMapNode = JsonNodeFactory.instance.arrayNode();
+    nonStringMapNode.add(JsonNodeFactory.instance.numberNode(1));
+    nonStringMapNode.add(JsonNodeFactory.instance.numberNode(1));
+    ArrayNode nonStringMapArrayNode = JsonNodeFactory.instance.arrayNode();
+    nonStringMapArrayNode.add(nonStringMapNode);
+    nonStringMapSchema.addProp("connect.default", nonStringMapArrayNode);
+
+    org.apache.avro.Schema avroSchema = org.apache.avro.SchemaBuilder
+        .record(AvroData.DEFAULT_SCHEMA_NAME).namespace(AvroData.NAMESPACE) // default values
+        .fields()
+        .name("int8").doc("int8 field").type(int8Schema).withDefault(2)
+        .name("int16").doc("int16 field").type(int16Schema).withDefault(12)
+        .name("int32").doc("int32 field").type(int32Schema).withDefault(12)
+        .name("int64").doc("int64 field").type(int64Schema).withDefault(12L)
+        .name("float32").doc("float32 field").type(float32Schema).withDefault(12.2f)
+        .name("float64").doc("float64 field").type(float64Schema).withDefault(12.2)
+        .name("boolean").doc("bool field").type(boolSchema).withDefault(true)
+        .name("string").doc("string field").type(stringSchema).withDefault("foo")
+        .name("bytes").doc("bytes field").type(bytesSchema).withDefault(ByteBuffer.wrap("foo".getBytes()))
+        .name("array").type(arraySchema).withDefault(Arrays.asList("a", "b", "c"))
+        .name("map").type(mapSchema).withDefault(Collections.singletonMap("field", 1))
+        .endRecord();
+    org.apache.avro.generic.GenericRecord avroRecord
+        = new org.apache.avro.generic.GenericRecordBuilder(avroSchema)
+        .set("int8", 2)
+        .set("int16", 12)
+        .set("int32", 12)
+        .set("int64", 12L)
+        .set("float32", 12.2f)
+        .set("float64", 12.2)
+        .set("boolean", true)
+        .set("string", "foo")
+        .set("bytes", ByteBuffer.wrap("foo".getBytes()))
+        .set("array", Arrays.asList("a", "b", "c"))
+        .set("map", Collections.singletonMap("field", 1))
         .build();
+
+    org.apache.avro.generic.GenericRecord convertedAvroRecord = (org.apache.avro.generic
+        .GenericRecord) convertedRecord;
+    assertSchemaEquals(avroSchema, convertedAvroRecord.getSchema());
+    assertSchemaEquals(avroRecord.getSchema(), convertedAvroRecord.getSchema());
+
+    // This doesn't work because the long field's default value is an integer
+    //assertEquals(avroRecord, convertedRecord);
```

Comment:
The `testFromConnectComplex` does not attempt to set the default on the Connect schema, so therefore the Avro schema has no default. Per the comment on the previous line, the Avro schema parses the default value as a JSON number (always an integer because of the value), so the Avro schema's default is incorrectly an integer rather than a long (correct).

Reply:
Maybe would it help to set Long.MAX_VALUE as the default?

Reply:
Well, that works but then the float default values are parsed via JSON as double values. :-(